### ISSUE TITLE
Strange scroll behaviour of file with charts

### ISSUE THREAD

**Issue by moose-x86** [phibr0/obsidian-charts#38]
Hi,
I have a problem with strange behavior when scrolling up file where charts are embedded. I'm scrolling to bottom of the file everything is fine, but when I scroll up charts disappear and also scroll behaves strangely. I see this issue only in file with charts. Tried to disable all other plugins, but there is no change. I attached small video to visualize problem.

![Peek 2022-01-17 10-57](https://user-images.githubusercontent.com/24254074/149748870-673f2ca0-336d-4410-933e-7a265bde43ed.gif)

**Comment by phibr0** [comment]
Could you check the console for errors? (CTRL + Shift + I)
The scrolling issue comes from Obsidian unloading contents not in the view and not knowing the size of the chart if its not rendered. But it should reappear correctly when scrolling up again.

**Comment by moose-x86** [comment]
When I add side by side pane, charts are rendering. There is a small glitch when scrolling, but at least charts are not disappearing. On mobile version charts are rendering, so it looks like it is somehow connected with width of pane(maybe).

![Peek 2022-01-17 11-35](https://user-images.githubusercontent.com/24254074/149754170-deda4b00-8afb-4d0e-b53b-3cfce579de1f.gif)

From charts I think there is only one error

![image](https://user-images.githubusercontent.com/24254074/149754619-fd8ab6b1-18e6-44a1-a1f5-79dc68df4842.png)
``

**Comment by moose-x86** [comment]
Also attaching file [data.md](https://github.com/phibr0/obsidian-charts/files/7881621/data.md) on which I see issue. System which I test is Debian 11 and Obsidian is appImage.

**Comment by phibr0** [comment]
Does this still occur? I can't reproduce this

**Comment by moose-x86** [comment]
yes, it still happens. On latest charts and obsidian 0.13.33 I see this problem, maybe a little bit less frequently(hard to say). i noticed that on default theme it works a little bit better but on Obuntu theme it looks more bugy. Maybe try to change a theme.

**Comment by moose-x86** [comment]
Oh and one funny thing. Legened disappered from charts after update. Now I need to set legend: true in order to have it. Is this a feature? In manual there is no mention of it.

**Comment by moose-x86** [comment]
One more thing. It is interesting that problem occurs in preview mode. In live preview it works almost ok(small glitch when scrolling, but charts are rendering in corret way.)

**Comment by phibr0** [comment]
> Oh and one funny thing. Legened disappered from charts after update. Now I need to set legend: true in order to have it. Is this a feature? In manual there is no mention of it.

will be fixed

**Comment by moose-x86** [comment]
> > One more thing. It is interesting that problem occurs in preview mode. In live preview it works almost ok(small glitch when scrolling, but charts are rendering in correct way.)
> 
> I might (?) have a fix for that, but as I cannot reproduce the Issue, I dont know if it works.

Maybe I could provide you with some configuration files? If needed. If there will be some kind patch I can patch code manually a test that.

**Comment by moose-x86** [comment]
Also one interesting thing is that sometimes font changes color I don't know if this is related. i see this on android also. 
![image](https://user-images.githubusercontent.com/24254074/159922096-a044b3bc-53e9-4223-90bf-06b1e2220141.png)

**Comment by moose-x86** [comment]
And one more thing I noticed today that changing from dark mode to light mode makes this bug go away for some time. When it returns changing back to dark mode also helps for some time. (meaning ths scroll issue.)

**Comment by moose-x86** [comment]
Hi. Any updates?

**Comment by moose-x86** [comment]
I noticed that changing option `Readable line length` to On in Editor settings, fixes issue. So I believe that this might be connected.
